iroh-api/benches/get.rs

```diff
@@ -0,0 +1,122 @@
+        group.bench_with_input(
+            BenchmarkId::new("file_size", *file_size as u64),
+            &path,
+            |b, path| {
+                let dir = tempfile::tempdir().unwrap();
+                let executor = Runtime::new().unwrap();
+                let server_addr = Addr::new_mem();
+                let client_addr = server_addr.clone();
+                let rpc_client = RpcClientConfig {
+                    store_addr: Some(client_addr),
+                    ..Default::default()
+                };
+
+                let config = StoreConfig {
+                    path: dir.path().join("db"),
+                    rpc_client: rpc_client.clone(),
+                    metrics: MetricsConfig::default(),
+                };
+                let (_task, client, resolver) = executor.block_on(async {
+                    let store = Store::create(config).await.unwrap();
+                    let task = executor.spawn(async move {
+                        iroh_store::rpc::new(server_addr, store).await.unwrap()
+                    });
+                    // wait for a moment until the transport is setup
+                    // TODO: signal this more clearly
+                    tokio::time::sleep(std::time::Duration::from_millis(500)).await;
+                    let client = Client::new(rpc_client).await.unwrap();
+                    let content_loader = FullLoader::new(
+                        client.clone(),
+                        FullLoaderConfig {
+                            http_gateways: Vec::new(),
+                            indexer: None,
+                        },
+                    )
+                    .unwrap();
+                    let resolver = Resolver::new(content_loader);
+
+                    (task, client, resolver)
+                });
+
+                let root = executor.block_on(async {
+                    let api = Api::from_client_and_resolver(client.clone(), resolver.clone());
+                    let entry = UnixfsEntry::from_path(
+                        path,
+                        UnixfsConfig {
+                            wrap: false,
+                            chunker: Some(ChunkerConfig::Fixed(DEFAULT_CHUNKS_SIZE)),
+                        },
+                    )
+                    .await
+                    .unwrap();
+                    let mut stream = api.add_stream(entry).await.unwrap();
+                    let mut cids: Vec<Cid> = Vec::new();
+                    while let Some(prog) = stream.next().await {
+                        let (cid, _) = prog.unwrap();
+                        cids.push(cid);
+                    }
+                    *cids.last().unwrap()
+                });
+
+                b.to_async(&executor).iter(|| {
+                    let api = Api::from_client_and_resolver(client.clone(), resolver.clone());
+                    let root = IpfsPath::from_cid(root);
+                    let mut dev_null = tokio::io::sink();
```

Comment:
You are timing the creation and dropping of the `Api` here and other here.  You probably want to use something like `.iter_batched()` with a setup function.  At the end of the `async move` block you then probably should return this all as a tuple so it gets dropped outside of the bench function.